libc/src/__support/intrusive_list.h

```diff
@@ -0,0 +1,65 @@
+//===-- Intrusive queue implementation. -------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// An intrusive list that implements the insque and remque semantics.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_LIBC_SRC___SUPPORT_INVASIVE_QUEUE_H
+#define LLVM_LIBC_SRC___SUPPORT_INVASIVE_QUEUE_H
+
+#include "common.h"
+
+namespace LIBC_NAMESPACE {
+namespace internal {
+
+class IntrusiveList {
```

Comment:
Can/Should we generalize this into
```c++
template <typename T, T* T::*Next, T* T::*Prev>
```
See example at https://github.com/facebookexperimental/libunifex/blob/f12ec5decb9018b765aba048111b7fa10715aece/include/unifex/detail/intrusive_list.hpp#L23

Reply:
Does other parts of libc need the generalized intrusive list implementation? If so we can generalize it.

Reply:
I am good with current implementation. We can generalize this later if there are other uses.

Reply:
insque looks a lot like LIST (`man 7 queue`)